src/psyclone/psyir/transformations/omp_target_trans.py

```diff
@@ -86,12 +87,34 @@ class OMPTargetTrans(RegionTrans):
     '''
     excluded_node_types = (CodeBlock, )
 
+    def validate(self, node, options=None):
+        # pylint: disable=signature-differs
+        '''
+        Check that we can safely enclose the supplied node or list of nodes
+        within OpenACC kernels ... end kernels directives.
+
+        :param node: the PSyIR node or nodes to enclose in the OpenMP
+                      target region.
+        :type node: List[:py:class:`psyclone.psyir.nodes.Node`]
+        :param options: a dictionary with options for transformations.
+        :type options: Optional[Dict[str, Any]]
+        '''
+        node_list = self.get_node_list(node)
+        super().validate(node, options)
+        for node in node_list:
+            for call in node.walk(Call):
+                if not call.is_available_on_device():
+                    raise TransformationError(
+                        f"'{call.routine.name}' is not available on the "
+                        f"accelerator device, and therefore it can not "
+                        f"be enclosed in a OMP target region.")
```

Comment:
"be called from within an OMP target region."

Reply:
Done